diesel/src/query_builder/bind_collector.rs

```diff
@@ -46,6 +46,23 @@ pub trait BindCollector<'a, DB: TypeMetadata>: Sized {
     }
 }
 
+/// Bind data from a BindCollector, that can be moved across threads
+pub trait MoveableBindData: Send + 'static {}
+
+/// A movable version of the bind collector which allows it to be extracted, moved and refilled.
+///
+/// This is mostly useful in async context where bind data needs to be moved across threads.
+pub trait MoveableBindCollector<DB: TypeMetadata> {
+    /// The movable bind data of this bind collector
+    type BindData: MoveableBindData;
```

Comment:
Can we just have `type BindData: Send + 'static;` instead here? Or do we need that `MovableBindData` trait for something else?

Reply:
It was a bit simplifying diesel_async trait bounds with `MD: MoveableBindData` instead of `MD: Send + 'static`. I went ahead and removed the `MoveableBindData` trait as suggested since it doesn't have much added value indeed.